src/NuGet.Services.AzureSearch/Analysis/PackageIdCustomAnalyzer.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Azure.Search.Models;
+
+namespace NuGet.Services.AzureSearch
+{
+    /// <summary>
+    /// Support for NuGet style package id searching.
+    /// See <see cref="PackageIdCustomTokenFilter"/> for more information.
+    /// </summary>
+    public static class PackageIdCustomAnalyzer
```

Comment:
This replaces the old search's `IdentifierAnalyzer`, but has a few differences:

1. The old analyzer split on the symbols `. - , ; : \ * # ! ~ ~ + - ( ) [ ] { }`. The new analyzer splits on any non alpha-numeric characters. For example: `test_package` used to be tokenized as `test_package`, it will now tokenize as `test` and `package`
2. The old analyzer would split numerical values differently. For example: `abc123def` used to be tokenized as `abc` and `123def`, it will now tokenize as `abc`, `123`, and `def`
3. The old tokenizer had some hard-coded expansions:
    * `ef` into `ef` and `entity framework`
    * `uwp` into `uwp` and `universal windows platform`
    * `uap` into `uwp` and `universal windows platform`

Reply:
See [these tests](https://github.com/NuGet/NuGet.Services.Metadata/pull/487/files#diff-31417718a0bff8df26dbb142b06af505R32) for tokenization examples